### PR TITLE
fix(allow multiple data analysis type in mip)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Currently cg upload -c fails for mip-dna cases that have multiple analysis types (WES;WGS) for the same case. however, the workflow supports the use of multiple data analysis types for the same case.

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/workflow/analysis.py [hunk 1]
```diff
@@ -712,7 +712,7 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
         analysis_types: set[str] = {
             link.sample.application_version.application.analysis_type for link in case.links
         }
-        if len(analysis_types) > 1:
```
Comment (islean) [inline cg/meta/workflow/analysis.py:715]: Won't this give an unclear return statement? It just pops the first analysis type in the list but if we have several we might need some extra logic

File: cg/meta/workflow/analysis.py [hunk 2]
```diff
@@ -712,7 +712,7 @@ def get_data_analysis_type(self, case_id: str) -> str | None:
...
+        if len(analysis_types) > 1 and self.workflow != Workflow.MIP_DNA:
```
Comment (islean) [inline cg/meta/workflow/analysis.py:715]: Won't this give an unclear return statement? It just pops the first analysis type in the list but if we have several we might need some extra logic

Comment (ChrOertlin) [reply cg/meta/workflow/analysis.py:715]: hmm I see... perhaps we should not even go into the function for ip dna but handle the function call based on the workflow?

Comment (islean) [reply cg/meta/workflow/analysis.py:715]: A bit uncertain but it is set in the RarediseaseParameters. Might need to check with some Raredisease developer to know the implications?

**Review by fevac — APPROVED** [review]
Looks good to me as a quick fix. However, this logic would make more sense during the ordering or the start step. If we don't handle multiple analyses types by the workflow, why even run them?

**Comment by ChrOertlin** [comment]
> Is this only to upload one case? If so I don't think this is worth introducing this extra logic layer

I don't know. From the infomration that I have gathered.
1. MIP can be run like this
2. Logic was introduced that now causes an error 27-5-24, which did not happen previously.
3. I don't know if this is an isolated incident or a more common occurence.

**Comment by henrikstranneheim** [comment]
I remember being very hesitant adding this feature in MIP. @jemten Is this allowed in raredisease? Given that MIP is nearing its end-of-life - is this important enough to fix @Clinical-Genomics/prodbioinfo ?

**Comment by Karl-Svard** [comment]
It's currently blocking a case in production from being uploaded, `comicwasp`. So would be great if it could be fixed shortly! 

In the past, these types of cases (WGS + WES) were analyzed as WES. This shouldn't have changed as far as I know

**Comment by ChrOertlin** [comment]
Ok, since this is a no longer supported feature I will close the PR.

**Pull request closed without merging**